### PR TITLE
chore: Test Improvements

### PR DESCRIPTION
**Pull request by glevco**
- ~~Add ability to pass a `HathorSettings` object to `CliBuilder.create_manager()` to help mocking settings on tests.~~
- Change way filepath is constructed on `_load_peer_id_pool()`, so it's calculated. Before, with the hardcoded path, running tests from any directory other then the project root would cause errors. This prevented running specific test files from an IDE.

Depends on https://github.com/HathorNetwork/hathor-core/pull/520.

_EDIT: I have rolled back the first change as I found a better way to do it and it's not necessary anymore._

### CODE REVIEW

**Review by jansegre — CHANGES_REQUESTED** [review]
I think this won't work quite as expected. I'll talk to you to see a way around it. Anyway, I the change proposed on this PR shouldn't be merged.

File: hathor/builder.py [hunk 1]
```diff
@@ -26,6 +26,8 @@
 from twisted.web import server
 from twisted.web.resource import Resource
 
+from hathor.conf.get_settings import HathorSettings as DefaultHathorSettings
```
Comment (msbrogli) [inline hathor/builder.py:29]: You don't need the get_settings.

File: hathor/builder.py [hunk 2]
```diff
@@ -362,7 +367,7 @@ def create_resources(self, args: Namespace) -> server.Site:
         )
         from hathor.websocket import HathorAdminWebsocketFactory, WebsocketStatsResource
 
-        settings = HathorSettings()
+        settings = DefaultHathorSettings()
```
Comment (msbrogli) [inline hathor/builder.py:370]: Why are you mixing different settings? Shouldn't all created objects use the same?

File: tests/unittest.py [hunk 3]
```diff
@@ -32,8 +32,14 @@ def shorten_hash(container):
     return container_type(h[-2:].hex() for h in container)
 
 
-def _load_peer_id_pool(file_path: str = 'tests/peer_id_pool.json') -> Iterator[PeerId]:
+def _load_peer_id_pool(file_path: Optional[str] = None) -> Iterator[PeerId]:
     import json
+
+    if file_path is None:
+        this_file_path = os.path.dirname(__file__)
```
Comment (msbrogli) [inline tests/unittest.py:39]: Maybe move it to a `get_default_peer_id_filepath(...)`?

**Review by msbrogli — CHANGES_REQUESTED** [review]

File: hathor/builder.py (hunk 1 above)
Comment (glevco) [reply hathor/builder.py]: I have rolled back that change as I found a better way to do it and it's not necessary anymore.

File: hathor/builder.py (hunk 2 above)
Comment (glevco) [reply hathor/builder.py]: I have rolled back that change as I found a better way to do it and it's not necessary anymore.

File: tests/unittest.py (hunk 3 above)
Comment (glevco) [reply tests/unittest.py]: Done!

**Comment by glevco** [comment]
> I think this won't work quite as expected. I'll talk to you to see a way around it. Anyway, I the change proposed on this PR shouldn't be merged.

@jansegre could you please re-review it? I have rolled back the `HathorSettings` change as I found a better way to do it and it's not necessary anymore.

**Review by msbrogli — APPROVED** [review]

File: hathor/builder.py (hunk 2 above)
Comment (msbrogli) [reply hathor/builder.py]: Ok!

File: tests/unittest.py (hunk 3 above)
Comment (msbrogli) [reply tests/unittest.py]: Thanks!

File: hathor/builder.py (hunk 1 above)
Comment (msbrogli) [reply hathor/builder.py]: Ok!

**Review by jansegre — APPROVED** [review]

**Pull request merged**